src/main/kotlin/org/rust/lang/core/psi/ext/RsTraitItem.kt

```diff
@@ -76,6 +76,20 @@ private val RsTraitItem.superTraits: Sequence<BoundElement<RsTraitItem>> get() {
     return bounds.mapNotNull { it.bound.traitRef?.resolveToBoundTrait }
 }
 
+val RsTraitItem.isSized: Boolean get() {
+    val stub = stub
+    if (stub != null) return implementedTrait?.flattenHierarchy.orEmpty().any { it.element.isSizedTrait }
+
+    val whereBounds =
+        whereClause?.wherePredList.orEmpty()
+            .filter { (it.typeReference?.typeElement as? RsBaseType)?.name == "Self" }
+            .flatMap { it.typeParamBounds?.polyboundList.orEmpty() }
+    val bounds = typeParamBounds?.polyboundList.orEmpty() + whereBounds
+    // ugly solution, the trait might be refer as std::marker::Sized, or in other way
+    // TODO: parse it properly
+    return bounds.any { it.text.contains(Regex("Sized")) && it.q == null }
```

Comment:
1. `it.bound.traitRef.resolveToTrait.isSizedTrait`
2. Hmm, `it.q == null` is a more complex stuff. We have to create new stub implementation for `RsPolybound` and write boolean flag `hasQ` to it. Example commits that add stubs ([1](https://github.com/intellij-rust/intellij-rust/commit/8513a547cdce4b898613d667094f2c4df6b8d3b1), [2](https://github.com/intellij-rust/intellij-rust/commit/625c54dfcb3bdbbbea3c1cb2e3783c3744db5ffd), [3](https://github.com/intellij-rust/intellij-rust/commit/00abf615f7c395e408b401902c7903859b4a5211))

Reply:
The question mark stub is important, but more the important thing is the RsTraitRef.resolveToBoundTrait method does not resolve KnownItems such as Sized or Clone, etc., I cannot found the cause. Looks like it worked when the check it.element.isSizedTrait was added.

Debugging leads me [here](https://github.com/intellij-rust/intellij-rust/blob/master/src/main/kotlin/org/rust/lang/core/resolve/ref/RsPathReferenceImpl.kt#L61)... in our case the element.parent is not RsPathExpr, so it returns nothing. I think the resolving should be fixed.

We can omit the question mark check for now, I think it is not relevant for traits.

Reply:
If the problem occurs in tests only, you probably have to add `@ProjectDescriptor(WithStdlibRustProjectDescriptor::class)` annotation to test method

Reply:
The is the root cause of the issue 2530, I fixed it using `it.text.contains(Regex("Sized"))` check, but it is wrong. I will try to find the commit that brokes resolution. It might take a while.

Reply:
Please, provide an exact test that shows where name resolution is broken. In the case of test contained in this PR it's enough to add `@ProjectDescriptor(WithStdlibRustProjectDescriptor::class)` to fix it.

Reply:
The hook is not taking into account types constructed from `Self`, for example `(Self, Self)` considered as not sized.